src/pal_platform.h

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2023 Advanced Micro Devices, Inc. All rights reserved.
+
+#ifndef PAL_PLATFORM_H
+#define PAL_PLATFORM_H
+
+#include "pal/pal_device.h"
+#include "pal/pal_utils.h"
+#include "platform.h"
+#include "runtime.h"
+
+#include <atomic>
+#include <functional>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include <pal.h>
+#include <palLib.h>
+#include <palPipeline.h>
+#include <palPlatform.h>
+#include <palQueue.h>
+
+#ifdef AnyDSL_runtime_HAS_LLVM_SUPPORT
+#include <llvm/IR/DataLayout.h>
+#include <llvm/IR/LLVMContext.h>
+#include <llvm/Linker/Linker.h>
+#include <llvm/Pass.h>
+#include <llvm/Passes/OptimizationLevel.h>
+#include <llvm/Support/SourceMgr.h>
+#include <llvm/Target/TargetMachine.h>
+#endif
+
+namespace llvm_utils {
+
+// Initializes llvm with amdgpu using the following commandline options:
```

Comment:
Where is the envvar set to this ? I don't see a matching change in the metarepo.

Reply:
Hmm. I think we may have missinterpreted the comment in HSAPlatform::emit_gcn():
https://github.com/AnyDSL/runtime/blob/36a5f3b7bc763497c22a7053230ea8a1012329b5/src/hsa_platform.cpp#L575
Is it actually just in there to set those variables quickly?

Reply:
You can set those code-gen variables via the environment variable `ANYDSL_LLVM_ARGS`, see:
https://anydsl.github.io/Device-Code-Generation-and-Execution.html#amdgpu-code-generation-optimization
They have been just added as a comment to the code.

Reply:
I see :+1: I've fixed the comment to reflect that better.